### PR TITLE
[EasyCore] Add helper traits for Symfony projects

### PR DESCRIPTION
**Pull request by DKeeper**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #342

### CODE REVIEW

File: packages/EasyCore/tests/AbstractTestCase.php [hunk 1]
```diff
@@ -39,4 +39,49 @@ protected function tearDown(): void
...
+     */
+    protected function setPrivatePropertyValue($object, string $property, $value): void
+    {
+        (function ($property, $value): void {
+            $this->{$property} = $value;
+        })->call($object, $property, $value);
+    }
+
+    /**
+     * Calls object's private method and returns it's result.
```
Comment (egor-dev) [inline packages/EasyCore/tests/AbstractTestCase.php:73]: `its`

File: packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace EonX\EasyCore\Tests\Bridge\Symfony\Traits;
+
+use EonX\EasyCore\Bridge\Symfony\Traits\TranslatorAwareTrait;
+use EonX\EasyCore\Tests\Bridge\Symfony\AbstractSymfonyTestCase;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class TranslatorAwareTraitTest extends AbstractSymfonyTestCase
+{
+    public function testSetEntityManagerSucceeds(): void
```
Comment (egor-dev) [inline packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php:13]: testSetTranslatorSucceeds

File: phpstan.neon [hunk 3]
```diff
@@ -83,6 +83,12 @@ parameters:
         - message: '#Call to an undefined method ApiPlatform\\Core\\DataPersister\\ContextAwareDataPersisterInterface\:\:(getSimpleDataPersisters|getDataPersisters)\(\)#'
           path: packages/EasyCore/src/Bridge/Symfony/ApiPlatform/DataPersister/TraceableChainSimpleDataPersister.php
 
+        - message: '#Parameter \#1 \$entityManager of method class@anonymous/EasyCore/tests/Bridge/Symfony/Traits/EntityManagerAwareTraitTest\.php\:15\:\:setEntityManager\(\) expects Doctrine\\ORM\\EntityManagerInterface, Mockery\\MockInterface given#'
```
Comment (egor-dev) [inline phpstan.neon:86]: Do we really need these? You've already added PHPDocs for the corresponding variables. Should be enough.

File: packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Bridge\Symfony\Doctrine;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+
+final class EntityManagerResolver
```
Comment (egor-dev) [inline packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php:10]: Please, add a unit test.

File: ecs.php [hunk 5]
```diff
@@ -149,6 +150,9 @@
             __DIR__ . '/packages/EasyErrorHandler/src/Bridge/Laravel/ExceptionHandler.php',
             __DIR__ . '/packages/EasyErrorHandler/tests/Bridge/Laravel/ExceptionHandlerTest.php',
         ],
+        ReturnAssignmentFixer::class => [
```
Comment (egor-dev) [inline ecs.php:153]: What is it for?

**Review by egor-dev — CHANGES_REQUESTED** [review]

File: ecs.php (hunk 5 above)
Comment (DKeeper) [reply ecs.php:153]:
```
vendor/bin/phpstan analyse packages/EasyCore/tests/Bridge/Symfony/Traits --error-format symplify
Note: Using configuration file /mnt/d/eonx/easy-monorepo/phpstan.neon.
 2/2 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  packages/EasyCore/tests/Bridge/Symfony/Traits/EntityManagerAwareTraitTest.php:21
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Parameter \#1 \$entityManager of method class@anonymous/EntityManagerAwareTraitTest\.php\:15\:\:setEntityManager\(\) expects Doctrine\\ORM\\EntityManagerInterface, Mockery\\MockInterface given#'
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php:21
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Parameter \#1 \$translator of method class@anonymous/TranslatorAwareTraitTest\.php\:15\:\:setTranslator\(\) expects Symfony\\Contracts\\Translation\\TranslatorInterface, Mockery\\MockInterface given#'
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

File: phpstan.neon (hunk 3 above)
Comment (DKeeper) [reply phpstan.neon:86]: `vendor/bin/phpstan analyse packages/EasyCore/tests/Bridge/Symfony/Traits --error-format symplify
Note: Using configuration file /mnt/d/eonx/easy-monorepo/phpstan.neon.
 2/2 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  packages/EasyCore/tests/Bridge/Symfony/Traits/EntityManagerAwareTraitTest.php:21
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Parameter \#1 \$entityManager of method class@anonymous/EntityManagerAwareTraitTest\.php\:15\:\:setEntityManager\(\) expects Doctrine\\ORM\\EntityManagerInterface, Mockery\\MockInterface given#'
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php:21
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Parameter \#1 \$translator of method class@anonymous/TranslatorAwareTraitTest\.php\:15\:\:setTranslator\(\) expects Symfony\\Contracts\\Translation\\TranslatorInterface, Mockery\\MockInterface given#'
 ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------`

**Review by roman-eonx — DISMISSED** [review]

File: packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php (hunk 2 above)
Comment (DKeeper) [reply packages/EasyCore/tests/Bridge/Symfony/Traits/TranslatorAwareTraitTest.php]: Done.

File: packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php (hunk 4 above)
Comment (DKeeper) [reply packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php:10]: Done.

**Review by roman-eonx — DISMISSED** [review]

**Review by egor-dev — DISMISSED** [review]

File: packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    private $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    /**
+     * @return \Doctrine\ORM\EntityManagerInterface
```
Comment (natepage) [inline packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php:23]: That's not needed because the method has a return type hint

**Review by natepage — CHANGES_REQUESTED** [review]

File: packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php (hunk 6 above)
Comment (DKeeper) [reply packages/EasyCore/src/Bridge/Symfony/Doctrine/EntityManagerResolver.php]: Done.

**Review by roman-eonx — APPROVED** [review]

**Review by egor-dev — APPROVED** [review]

**Pull request merged**